### DISCUSSION TITLE
Do not remove blank line at the end of file

### DISCUSSION THREAD

**Discussion by remusxb** [LazyVim/LazyVim#2277] [Q&A] [answered]
Hello 👋🏼 

I'm kinda new to nvim/lazyvim (2 days of playing with it) and I don't have much idea about the capabilities/settings.
One question is, how can I overwrite the behaviour of removing the blank line at the end of a file upon saving/:w ?

**Comment by dpetka2001** [comment, marked as answer]
That is the job of the formatter. Not LazyVim itself. For example with default LazyVim starter template, which only has `stylua` as formatter for `.lua` files, this only happens in `.lua` files. If you open a python file or C file it doesn't happen(because there are no formatters for these languages installed by default). So, depending on the formatter you use, you would have to see if it supports setting a rule about not removing last blank line.

**Reply by remusxb** [reply]
You're right. I previously had some formatters and LSPs set up, but couldn't manage to make them work, until I've installed LazyVim. After the instalment, i thought that lazy vim had some plugins/cmds which did that.

Thank you!
